From a3412db525f244a13efeb89e84e93d4044ca8756 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rapha=C3=ABl=20Hertzog?= <hertzog@debian.org>
Date: Thu, 23 Jul 2015 17:04:06 +0200
Subject: [PATCH 1/2] Restore correct behaviour for
 APT::Never-MarkAuto-Sections

It should mark dependencies of packages matching this setting as manually
installed. Right now it's marking the packages themselves and only when
they are part of an upper-level dependency tree.

Regression introduced in 50ef3344c3afaaf9943142906b2f976a0337d264.

Sponsored-by: Offensive Security
---
 apt-pkg/depcache.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apt-pkg/depcache.cc b/apt-pkg/depcache.cc
index 16282df..0b2f71d 100644
--- a/apt-pkg/depcache.cc
+++ b/apt-pkg/depcache.cc
@@ -1226,7 +1226,7 @@ bool pkgDepCache::MarkInstall(PkgIterator const &Pkg,bool AutoInst,
 	       continue;
 	    }
 	    // now check if we should consider it a automatic dependency or not
-	    if(InstPkg->CurrentVer == 0 && InstVer->Section != 0 && ConfigValueInSubTree("APT::Never-MarkAuto-Sections", InstVer.Section()))
+	    if(P.InstallVer->Section != 0 && ConfigValueInSubTree("APT::Never-MarkAuto-Sections", P.InstVerIter(*this).Section()))
 	    {
 	       if(DebugAutoInstall == true)
 		  std::clog << OutputInDepth(Depth) << "Setting NOT as auto-installed (direct "
-- 
2.1.4

